src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -95,8 +98,39 @@ open class KtlintPlugin : Plugin<Project> {
         // Add the args to enable verbose and debug mode.
         if (extension.verbose) runArgs.add("--verbose")
         if (extension.debug) runArgs.add("--debug")
+        addReporter(extension, runArgs)
     }
 
+    private fun addReporter(extension: KtlintExtension, runArgs: MutableSet<String>) {
+        if (isReportAvailable(extension.version)) {
+            runArgs.add("--reporter=${extension.reporter}")
+        }
+    }
+
+    private fun isReportAvailable(version: String): Boolean {
+        val versionsNumbers = version.split('.')
+        if (versionsNumbers[0].toInt() >= 1) {
+            return true
+        } else if (versionsNumbers[1].toInt() >= 9) {
+            return true
+        }
+
+        return false
+    }
+
+    private fun createCheckTaskOutputDir(target: Project, extension: KtlintExtension): File {
+        val reportsDir = File(target.buildDir, "reports/ktlint")
+        reportsDir.mkdirs()
+        return File(reportsDir, "ktlint.${getReportFileExtension(extension)}")
+    }
+
+    private fun getReportFileExtension(extension: KtlintExtension): String =
+            when(extension.reporter) {
+                "checkstyle" -> "xml"
+                "json" -> "json"
+                else -> "txt"
+            }
```

Comment:
Then `txt` will be used. Again I can introduce enum for this reporters and define there file extension.